mlflow/tracking/request_header/databricks_request_header_provider.py

```diff
@@ -0,0 +1,21 @@
+from mlflow.tracking.request_header.abstract_request_header_provider import RequestHeaderProvider
+from mlflow.utils import databricks_utils
+
+
+class DatabricksRequestHeaderProvider(RequestHeaderProvider):
+    def in_context(self):
+        return databricks_utils.is_in_cluster()
```

Comment:
Is the cluster ID guaranteed to be present in all contexts? I'm wondering if instead we should return `is_in_cluster() or is_in_databricks_notebook() or is_in_databricks_job()` to make sure that we still use a subset of headers even if the cluster ID can't be fetched

Reply:
Good idea!